core/src/preferences/PreferencesBackend.cpp

```diff
@@ -183,7 +183,7 @@ namespace ledger {
 
         std::string PreferencesBackend::createNewSalt(const std::shared_ptr<api::RandomNumberGenerator>& rng) {
             auto bytes = rng->getRandomBytes(128);
-            return std::string(std::begin(bytes), std::end(bytes));
+            return std::string(bytes.begin(), bytes.end());
         }
 
         void PreferencesBackend::setEncryption(
```

Comment:
setEncryption is called in resetEncryption, so we iterate throw the DB twice in worste case

Reply:
That happens if someone calls that function on a non-encrypted DB. I’ll add a check for that.

Reply:
I think if you refactor it a little bit, as I point in other comment, there will be no possibility to iterate twice.

Reply:
Sure, I will!